### PR TITLE
[SPARK-37864][SQL] Support vectorized read boolean values use RLE encoding with Parquet DataPage V2

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
Parquet v2 data page write Boolean Values use RLE encoding, when read v2 boolean type values it will throw exceptions as follows now:

```java
Caused by: java.lang.UnsupportedOperationException: Unsupported encoding: RLE
    at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.getValuesReader(VectorizedColumnReader.java:305) ~[classes/:?]
    at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.initDataReader(VectorizedColumnReader.java:277) ~[classes/:?]
    at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.readPageV2(VectorizedColumnReader.java:344) ~[classes/:?]
    at 
```

This PR extends the `readBooleans` and `skipBooleans` of `VectorizedRleValuesReader` to ensure that the above scenario can pass.

### Why are the changes needed?
Support Parquet v2 data page RLE encoding  for the vectorized read path

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Add new test case

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetEncodingSuite.scala [hunk 1]
```diff
@@ -182,4 +182,40 @@ class ParquetEncodingSuite extends ParquetCompatibilityTest with SharedSparkSess
       }
     }
   }
+
+  test("parquet v2 pages - rle encoding for boolean value columns") {
```
Comment (LuciferYang) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetEncodingSuite.scala:186]: Before this pr, this case will throw exception:
```
Caused by: java.lang.UnsupportedOperationException: Unsupported encoding: RLE
	at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.getValuesReader(VectorizedColumnReader.java:304) ~[classes/:?]
	at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.initDataReader(VectorizedColumnReader.java:276) ~[classes/:?]
	at org.apache.spark.sql.execution.datasources.parquet.VectorizedColumnReader.readPageV2(VectorizedColumnReader.java:343) ~[classes/:?]

```

File: sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java [hunk 2]
```diff
@@ -407,7 +426,7 @@ public void skipIntegers(int total) {
 
   @Override
   public void skipBooleans(int total) {
-    throw new UnsupportedOperationException("only skipIntegers is valid");
+    skipIntegers(total);
```
Comment (sunchao) [inline sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java:429]: nit: maybe create another method like `skipN` and have `skipIntegers` and `skipBooleans` share it? it's a bit confusing when first reading this.

**Review by sunchao — APPROVED** [review]
LGTM with one nit

File: sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java (hunk 2 above)
Comment (LuciferYang) [reply sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java:429]: OK ~

Comment (LuciferYang) [reply sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java]: @sunchao 7a9d4fd fix this

**Comment by sunchao** [comment]
Merged to master, thanks @LuciferYang !

**Comment by LuciferYang** [comment]
thanks @sunchao

**Pull request closed without merging**